main.js

```diff
@@ -45,79 +123,244 @@ function longestWord(sen) {
+        result === "o" ||
+        result === "i" ||
+        result === "u"
+      ) {
+        ans = ans + result.toUpperCase();
+      } else {
+        ans = ans + result;
+      }
+    }
+  }
+  return ans;
+}
 
 /* 12. Return a sum of all parameters entered regardless of the amount of numbers - NO ARRAYS
  * ex. addAll(2,5,6,7) === 20
  */
-function addAll() {}
-
-
+function addAll() {
+  let ans = 0;
+  for (let i = 0; i < arguments.length; i++) {
+    ans += arguments[i];
+  }
+  return ans;
+}
 
 /* 13. Pass in a number to loop up to and add all of the prime numbers. A prime number is a whole number greater than 1 whose only factors are 1 and itself
  * ex. sumAllPrimes(10) == 17
  */
-function sumAllPrimes() {}
-
-
+function sumAllPrimes() {
+  let sum = 0;
+  let arr = new Array(arguments[0] + 1);
+  arr.fill(true);
+  arr[0] = false;
+  arr[1] = false;
+  for (let i = 2; i * i <= arguments[0]; i++) {
+    if (arr[i]) {
+      for (let j = i * 2; j <= arguments[0]; j += i) {
+        arr[j] = false;
+      }
+    }
+  }
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i]) {
+      sum += i;
+    }
+  }
+  return sum;
+}
 
 /* 14. Remove from the array whatever is in the following arguments. Return the leftover numbers in an array
  * ex. seekAndDestroy([2, 3, 4, 6, 6, 'hello'], 2, 6) == [3, 4, 'hello']
  */
-function seekAndDestroy() {}
-
-
+function seekAndDestroy() {
+  let arr = arguments[0];
+  let dummy = new Array();
+  let ans = new Array();
+  for (let i = 1; i < arguments.length; i++) {
+    dummy.push(arguments[i]);
+  }
+  for (let i = 0; i < arr.length; i++) {
+    if (dummy.includes(arr[i]) === false) {
+      ans.push(arr[i]);
+    }
+  }
+  return ans;
+}
 
 /* 15. Some people are standing in a row in a park. There are trees between them which cannot be moved. Your task is to rearrange the people by their heights in a non-descending order without moving the trees.
  * ex. sortByHeight([-1, 150, 190, 170, -1, -1, 160, 180]) == [-1, 150, 160, 170, -1, -1, 180, 190]
  */
-function sortByHeight() {}
-
-
+function sortByHeight() {
+  let arr = new Array();
+  //copy all heights into an array
+  for (let i = 0; i < arguments[0].length; i++) {
+    if (arguments[0][i] !== -1) {
+      arr.push(arguments[0][i]);
+    }
+  }
+  arr.sort(); //sort
+  let j = 0;
+  //assigning the height to the positions
+  for (let i = 0; i < arguments[0].length; i++) {
+    if (arguments[0][i] !== -1) {
+      arguments[0][i] = arr[j];
+      j++;
+    }
+  }
+  return arguments[0];
+}
 
 /* 16. Find the missing letter in the passed letter range and return it. If all letters are present, return undefined
  * ex. missingLetters("abce") == "d"
  * ex. missingLetters("abcdefghjklmno") == "i"
  * ex. missingLetters("abcdefghijklmnopqrstuvwxyz") == undefined
  */
-function missingLetters() {}
-
-
+function missingLetters() {
+  let ans;
+  let first = arguments[0].charCodeAt(0);
+  for (let i = 0; i < arguments[0].length; i++) {
```

Comment:
Repeatedly using `arguments[0]` everywhere, try using for..in or for..of loop anywhere

Reply:
Done